framework/src/binary_manager/binary_manager_get_state.c

```diff
@@ -0,0 +1,64 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/***************************************************************************
+ * Included Files
+ ***************************************************************************/
+
+#include <debug.h>
+#include <string.h>
+#include <tinyara/binary_manager.h>
+#include <binary_manager/binary_manager.h>
+#include "binary_manager_internal.h"
+
+binmgr_result_type_e binary_manager_get_state(char *binary_name, int *state)
+{
+	int ret;
+	binmgr_request_t request_msg;
+	binmgr_getstate_response_t response_msg;
+
+	if (binary_name == NULL || strlen(binary_name) > BIN_NAME_MAX - 1) {
+		bmdbg("get_binary_info failed : invalid param.\n");
+		return BINMGR_INVALID_PARAM;
+	}
+
+	ret = binary_manager_set_request(&request_msg, BINMGR_GET_STATE, binary_name);
+	if (ret < 0) {
+		return BINMGR_COMMUNICATION_FAIL;
+	}
+
+	ret = binary_manager_send_request(&request_msg);
+	if (ret < 0) {
+		bmdbg("Failed to send request msg %d\n", ret);
+		return BINMGR_COMMUNICATION_FAIL;
+	}
+
+	ret = binary_manager_receive_response(&response_msg, sizeof(binmgr_getstate_response_t));
+	if (ret < 0) {
+		bmdbg("Failed to receive response msg %d\n", ret);
+		return BINMGR_COMMUNICATION_FAIL;
+	}
+
+	if (response_msg.result == BINMGR_OK) {
+		*state = response_msg.state;
+	} else {
+		bmdbg("Binary manager getstate FAIL %d\n", response_msg.result);
+	}
+
+	return response_msg.result;
```

Comment:
Return type should be one of binmgr_result_type_e
```
	if (response_msg.result == BINMGR_OK) {
		*state = response_msg.state;
	} else {
		bmdbg("Binary manager getstate FAIL %d\n", response_msg.result);
                return BINMGR_OPERATION_FAIL; // or some other errors.
	}

	return BINMGR_OK;
}
```
so caller can use below code. (it just example)
```
if (binary_manager_get_state(path, &state) == BINMGR_OK) {
    switch (state) {
        case A:
        ...
    }
}
```

Reply:
Framework just sends commands and receives result from binary manager.
So it just returns result received from binary manager here because binary manager has all actual data and does operations on them.

Reply:
Let's think about why framework is necessary.
Framework is kind of abstracted and mixed layer of lower code to make application more easier.
So, usually it doesn't use whole of code in lower layer, and take some of lower codes.
(For example, camera framework takes most of camera code, and some of sound code, some of graphic code)
So with this framework, application doesn't have to care about **ANY OF** lower layer's code.
But with this framework, application should refer to both of framework and bm driver.

That's why I commented above, (https://github.com/Samsung/TizenRT/pull/4486#discussion_r500689499) refining is necessary.

Anyway your function's type is binmgr_response_result_type, it's not response_msg.result, right?

Reply:
A response_msg.result has binmgr_response_result_type type also so it returns  response_msg.result here.
Anyway I agree with your opinion. I will modify them with another commit.